packages/common/infra/src/livedata/index.ts

```diff
@@ -268,6 +269,38 @@ export class LiveData<T = unknown> implements InteropObservable<T> {
     this.upstreamSubscription?.unsubscribe();
   }
 
+  /**
+   * flatten the livedata
+   *
+   * ```
+   * new LiveData(new LiveData(0)).flat() // LiveData<number>
+   * ```
+   *
+   * ```
+   * new LiveData([new LiveData(0)]).flat() // LiveData<number[]>
+   * ```
+   */
+  flat(): T extends LiveData<infer P>
+    ? LiveData<P>
+    : T extends LiveData<infer P>[]
+      ? LiveData<P[]>
+      : never {
+    return LiveData.from(
+      this.pipe(
+        switchMap(v => {
+          if (v instanceof LiveData) {
+            return v as LiveData<any>;
+          } else if (v instanceof Array) {
```

Comment:
I think `Array.isArray` is preferred over `instanceof Array`

Reply:
done